src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -538,6 +544,60 @@ func (c controller) addRouterCAToClusterCA() bool {
 
 }
 
+// TODO: remove
+type OLMOperator struct {
+	name      string
+	namespace string
+	enabled   bool
+	status    string
+}
+
+// TODO: remove
+var olms = map[string]OLMOperator{
+	"lso": {
+		"local-storage-operator.4.6.0-202101300140.p0", "openshift-local-storage", true, "",
+	},
+}
+
+func (c controller) getOLMOperators() ([]OLMOperator, error) {
+	ret := []OLMOperator{}
+	operators, err := c.ic.GetClusterOLM(context.TODO())
+	if err != nil {
+		c.log.WithError(err).Warningf("Failed to connect to assisted service")
+		return ret, err
+	}
+	for _, operator := range *operators {
+		if *operator.Enabled && operator.Status != "Succeeded" {
+			ret = append(ret, olms[string(operator.OperatorType)])
+		}
+	}
+	return ret, nil
+}
+
+// TODO: remove
+func updateStatus(olm OLMOperator, phase string) {
+	olm.status = phase
+	olms[olm.name] = olm
+}
+
+func (c controller) waitForOLMOperators() bool {
+	c.log.Infof("Checking OLM operators")
+	opeartors, _ := c.getOLMOperators()
+	if len(opeartors) == 0 {
+		return true
+	}
+	for _, olm := range opeartors {
+		csv, err := c.kc.GetCSV(olm.namespace, olm.name)
+		if err != nil {
+			c.log.WithError(err).Warnf("Failed to get %s", olm.name)
```

Comment:
Shouldn't you `continue` the execution in case of error? I am not sure that the `csv` object could be considered correct, it might be even nil, in which case the next call to get the `csv.Status.Phase` should cause a panic. WDYT?

Reply:
Good catch, thanks. Maybe we can actually update the status with some error state and not `continue`. Let's see how the end design with status field will look like.